packages/dds/merge-tree/src/zamboni.ts

```diff
@@ -126,8 +133,9 @@ function scourNode(node: IMergeBlock, holdNodes: IMergeNode[], mergeTree: MergeT
 		if (childNode.isLeaf()) {
 			const segment = childNode;
 			if (segment.segmentGroups.empty) {
-				if (segment.removedSeq !== undefined) {
-					if (segment.removedSeq > mergeTree.collabWindow.minSeq) {
+				const removalInfo = toRemovalInfo(segment);
+				if (removalInfo !== undefined) {
+					if (!seqLTE(removalInfo.removedSeq, mergeTree.collabWindow.minSeq)) {
```

Comment:
there might have been a bug here if a segment was remove shortly after insert, so the insert put it on the pending list, and removed seq was unassigned. very corner case, but not impossible. this change would fix it.